pkg/check/util.go

```diff
@@ -0,0 +1,76 @@
+package check
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb-tools/pkg/utils"
+)
+
+// MySQLVersion represents MySQL version number.
+type MySQLVersion [3]uint
+
+// version format:
+// mysql        5.7.18-log
+// mariadb      5.5.50-MariaDB-1~wheezy
+// percona      5.7.19-17-log
+// aliyun rds   5.7.18-log
+// aws rds      5.7.16-log
+// ref: https://dev.mysql.com/doc/refman/5.7/en/which-version.html
+
+// v is mysql version in string format.
+func toMySQLVersion(v string) MySQLVersion {
+	tmp := strings.Split(v, "-")
+	tmp = strings.Split(tmp[0], ".")
+	if len(tmp) != 3 {
+		log.Warnf("invalid version %s", v)
+		return [3]uint{0, 0, 0}
+	}
+	version := [3]uint{}
+	for i := range tmp {
+		v, err := strconv.ParseUint(tmp[i], 10, 64)
+		if err != nil {
+			log.Warnf("invalid version %s", v)
```

Comment:
`log.Warnf("invalid version %s", tmp[i])` ?

Reply:
output complete string

Reply:
`v` become new variable returned by `strconv.ParseUint` (an uint64)？
maybe we need a `v2` or another

Reply:
we need to output complete string